src/protocol/resp/src/parse/error.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023 Pelikan Foundation LLC.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use std::borrow::Cow;
+use std::fmt;
+
+use bstr::BStr;
+use thiserror::Error;
+
+#[derive(Error)]
+pub enum ParseError<'a> {
+    #[error("ran out of data unexpectedly")]
+    Incomplete,
+    #[error("expected {:?}, got {:?} instead", BStr::new(.expected), BStr::new(.found))]
+    InvalidLiteral {
+        expected: &'static [u8],
+        found: Cow<'a, [u8]>,
+    },
+    #[error("expected a number, got {:?} instead", BStr::new(.0))]
```

Comment:
I suggest using quotes around the string value to make it more readable to the user.